homeassistant/components/mqtt/subscription.py

```diff
@@ -101,8 +101,9 @@ def async_prepare_subscribe_topics(
     sets of topics. Every call to async_subscribe_topics must always
     contain _all_ the topics the subscription state should manage.
     """
-    current_subscriptions = new_state if new_state is not None else {}
-    new_state = {}
+    current_subscriptions: dict[str, EntitySubscription]
+    current_subscriptions = sub_state if sub_state is not None else {}
+    new_state: dict[str, EntitySubscription] = {}
```

Comment:
Could this be called `sub_state` here too is is `new_state` something different entirely?

Reply:
done